ros/lib/utils.py

```diff
@@ -78,19 +90,58 @@ def user_data_from_identity(identity):
     return identity['user']
 
 
-def validate_ros_payload(is_ros, cloud_provider):
-    """
-    Validate ros payload.
-    :param is_ros: is_ros boolean flag
-    :param cloud_provider: cloud provider value
-    :return: True if cloud_provider is not none and is_ros flag is set to true, False otherwise.
-    """
-    return True if is_ros and cloud_provider is not None else False
+def get_payload_metadata(payload_data):
+    caller_prefix = payload_data[0].prefix  # Fetching prefix from class instance
+    msg = payload_data[1]
+    try:
+        if caller_prefix == 'ENGINE RESULTS':
+            inventory_id = msg["input"]["host"]["id"]
+            is_ros = msg["input"]["platform_metadata"].get("is_ros")
+            cloud_provider = msg["results"]["system"]["metadata"].get('cloud_provider')
+            account = msg['input']['host']['account']
+            org_id = msg['input']['platform_metadata']['org_id']
+        else:
+            inventory_id = msg['host']['id']
+            event_type = msg.get('type')
+            # exception for inventory update events
+            # is_ros flag is unexpected for the same
+            if event_type == 'updated':
+                is_ros = True
```

Comment:
I don't like the idea of setting `is_ros` flag here as at this level we are not sure whether payload related to system which already ROS enabled.

Reply:
My 2 cents:

1. As mentioned above, any ROS system record will only be updated if found with `update_system_record` method
2. This is temporary, we should ideally get system_metadata in kafka msg, once that is fixed we can remove this if block

So, I think this should be fine, for now.